### PR TITLE
Add `executable_name` example

### PR DESCRIPTION
**Pull request by maxwellE**
Related to #96.

Adds `executable_name` to `//examples/ios_app/Example`.

### CODE REVIEW

File: examples/ios_app/Example/BUILD [hunk 1]
```diff
@@ -14,6 +14,7 @@ ios_application(
     app_icons = glob(["Assets.xcassets/AppIcon.appiconset/**"]),
     bundle_id = "io.buildbuddy.example",
     bundle_name = "Example",
+    executable_name = "Examnple_ExecutableName",
```
Comment (maxwellE) [inline examples/ios_app/Example/BUILD:17]: @brentleyjones I am not sure this is working, unless the generated `Info.plist` just has this set.

Comment (brentleyjones) [reply examples/ios_app/Example/BUILD]: Correct. We need to add support for setting `EXECUTABLE_NAME` based on this value.

**Comment by maxwellE** [comment]
Likely this is not being set on the info plist

Screen Shot 2022-07-11 at 2 42 54 PM

File: examples/ios_app/Example/BUILD (hunk 1 above)
Comment (brentleyjones) [reply examples/ios_app/Example/BUILD]: Ahh, or rules_apple needs to adjust the Info.plist. What does a `bazel buld` produce inside the `.ipa`?

Comment (maxwellE) [reply examples/ios_app/Example/BUILD]: Info plist is correct:

```
	<key>CFBundleExecutable</key>
	<string>Example_ExecutableName</string>
```

File: test/internal/target/process_top_level_properties_tests.bzl [hunk 2]
```diff
@@ -87,7 +94,8 @@ def _bundle_info(
         bundle_extension,
         bundle_name,
         minimum_deployment_os_version,
-        product_type):
+        product_type,
+        executable_name):
```
Comment (brentleyjones) [inline test/internal/target/process_top_level_properties_tests.bzl:98]: Can you move this up to be alphabetical?

File: test/internal/target/process_top_level_properties_tests.bzl [hunk 3]
```diff
@@ -76,6 +82,7 @@ process_top_level_properties_test = unittest.make(
         "target_files": attr.string_list(mandatory = True),
         "target_name": attr.string(mandatory = True),
         "tree_artifact_enabled": attr.bool(mandatory = True),
+        "expected_executable_name": attr.string(mandatory = True),
```
Comment (brentleyjones) [inline test/internal/target/process_top_level_properties_tests.bzl:85]: Can you move this up to be alphabetical?

File: test/internal/target/process_top_level_properties_tests.bzl [hunk 4]
```diff
@@ -164,6 +176,7 @@ def process_top_level_properties_test_suite(name):
         expected_product_name = "binary",
         expected_product_type = "com.apple.product-type.tool",
         expected_build_settings = {},
+        expected_executable_name = "binary",
```
Comment (brentleyjones) [inline test/internal/target/process_top_level_properties_tests.bzl:179]: Can you move this up to be alphabetical?

File: test/internal/target/process_top_level_properties_tests.bzl [hunk 5]
```diff
@@ -201,7 +216,9 @@ def process_top_level_properties_test_suite(name):
         expected_product_type = "com.apple.product-type.application",
         expected_build_settings = {
             "PRODUCT_BUNDLE_IDENTIFIER": "com.example.flagship",
+            "EXECUTABLE_NAME": "executable_name",
```
Comment (brentleyjones) [inline test/internal/target/process_top_level_properties_tests.bzl:219]: Can you move this up to be alphabetical?

File: xcodeproj/internal/top_level_targets.bzl [hunk 6]
```diff
@@ -109,9 +110,11 @@ def process_top_level_properties(
             )
 
         build_settings["PRODUCT_BUNDLE_IDENTIFIER"] = bundle_info.bundle_id
+        build_settings["EXECUTABLE_NAME"] = executable_name
```
Comment (brentleyjones) [inline xcodeproj/internal/top_level_targets.bzl:113]:
```suggestion
        build_settings["EXECUTABLE_NAME"] = executable_name
        build_settings["PRODUCT_BUNDLE_IDENTIFIER"] = bundle_info.bundle_id
```

File: xcodeproj/internal/top_level_targets.bzl [hunk 7]
```diff
@@ -89,6 +89,7 @@ def process_top_level_properties(
         product_name = bundle_info.bundle_name
         product_type = bundle_info.product_type
         minimum_deployment_version = bundle_info.minimum_deployment_os_version
+        executable_name = bundle_info.executable_name
```
Comment (brentleyjones) [inline xcodeproj/internal/top_level_targets.bzl:92]:
```suggestion
        executable_name = bundle_info.executable_name
        minimum_deployment_version = bundle_info.minimum_deployment_os_version
        product_name = bundle_info.bundle_name
        product_type = bundle_info.product_type
```

File: xcodeproj/internal/top_level_targets.bzl [hunk 8]
```diff
@@ -109,9 +110,11 @@ def process_top_level_properties(
             )
 
         build_settings["PRODUCT_BUNDLE_IDENTIFIER"] = bundle_info.bundle_id
+        build_settings["EXECUTABLE_NAME"] = executable_name
     else:
         product_name = target_name
         minimum_deployment_version = None
+        executable_name = target_name
```
Comment (brentleyjones) [inline xcodeproj/internal/top_level_targets.bzl:117]:
```suggestion
        minimum_deployment_version = None
        executable_name = target_name
        product_name = target_name
```

File: xcodeproj/internal/top_level_targets.bzl [hunk 9]
```diff
@@ -138,6 +141,7 @@ def process_top_level_properties(
         minimum_deployment_os_version = minimum_deployment_version,
         product_name = product_name,
         product_type = product_type,
+        executable_name = executable_name,
```
Comment (brentleyjones) [inline xcodeproj/internal/top_level_targets.bzl:144]:
```suggestion
        executable_name = executable_name,
        minimum_deployment_os_version = minimum_deployment_version,
        product_name = product_name,
        product_type = product_type,
```

File: xcodeproj/internal/top_level_targets.bzl [hunk 10]
```diff
@@ -108,10 +109,12 @@ def process_top_level_properties(
                 path = bundle_path,
             )
 
+        build_settings["EXECUTABLE_NAME"] = executable_name
         build_settings["PRODUCT_BUNDLE_IDENTIFIER"] = bundle_info.bundle_id
     else:
-        product_name = target_name
         minimum_deployment_version = None
+        executable_name = target_name
```
Comment (brentleyjones) [inline xcodeproj/internal/top_level_targets.bzl:116]: One more (sorry 😅)

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**